src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests_Attributes_FullyQualifiedName.cs

```diff
@@ -0,0 +1,1159 @@
+        var runResult = driver.GetRunResult().Results[0];
+
+        Assert.False(runResult.TrackedSteps.ContainsKey("result_ForAttributeWithMetadataName"));
+    }
+
+    [Fact]
+    public void DoNotFindAttributeOnTopLevelClass_WhenSearchingSimpleName2()
+    {
+        var source = @"
+[N1.X]
+class C1 { }
+[N2.X]
+class C2 { }
+
+namespace N1
+{
+    class XAttribute : System.Attribute { }
+}
+
+namespace N2
+{
+    class XAttribute : System.Attribute { }
+}
+";
+        var parseOptions = TestOptions.RegularPreview;
+        Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+
+        Assert.Single(compilation.SyntaxTrees);
+
+        var generator = new IncrementalGeneratorWrapper(new PipelineCallbackGenerator(ctx =>
+        {
+            var input = ctx.ForAttributeWithMetadataName<ClassDeclarationSyntax>("X");
+            ctx.RegisterSourceOutput(input, (spc, node) => { });
+        }));
+
+        GeneratorDriver driver = CSharpGeneratorDriver.Create(new ISourceGenerator[] { generator }, parseOptions: parseOptions, driverOptions: new GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps: true));
+        driver = driver.RunGenerators(compilation);
+        var runResult = driver.GetRunResult().Results[0];
+
+        Assert.False(runResult.TrackedSteps.ContainsKey("result_ForAttributeWithMetadataName"));
+    }
+
+    [Fact]
+    public void FindCorrectAttributeOnTopLevelClass_WhenSearchingForClassDeclaration2()
+    {
+        var source = @"
+[N1.X]
+class C1 { }
+[N2.X]
+class C2 { }
+
+namespace N1
+{
+    class XAttribute : System.Attribute { }
+}
+
+namespace N2
+{
+    class XAttribute : System.Attribute { }
+}
+";
+        var parseOptions = TestOptions.RegularPreview;
+        Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+
+        Assert.Single(compilation.SyntaxTrees);
+
+        var generator = new IncrementalGeneratorWrapper(new PipelineCallbackGenerator(ctx =>
+        {
+            var input = ctx.ForAttributeWithMetadataName<ClassDeclarationSyntax>("N2.XAttribute");
+            ctx.RegisterSourceOutput(input, (spc, node) => { });
+        }));
+
+        GeneratorDriver driver = CSharpGeneratorDriver.Create(new ISourceGenerator[] { generator }, parseOptions: parseOptions, driverOptions: new GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps: true));
+        driver = driver.RunGenerators(compilation);
+        var runResult = driver.GetRunResult().Results[0];
+
+        Assert.Collection(runResult.TrackedSteps["result_ForAttributeWithMetadataName"],
+            step => Assert.True(step.Outputs.Single().Value is ClassDeclarationSyntax { Identifier.ValueText: "C2" }));
+    }
+
+    [Fact]
+    public void FindAssemblyAttribute1()
+    {
+        var source = @"
+using System;
+[assembly: CLSCompliant(true)]
+";
```

Comment:
Consider as an idea the following which would give us more coverage on the variations on how an attribute could be written.

```suggestion
    [Theory]
    [InlineData("CLSCompliant(true)"]
    [InlineData("CLSCompliantAttribute(true)"]
    [InlineData("System.CLSCompliant(true)"]
    public void FindAssemblyAttribute1(string attributeName)
    {
        var source = $$"""
using System;
[assembly: {{attributeName}}]
""";
```

Reply:
i like it :)

Reply:
Done!